zkevm-circuits/src/pi_circuit.rs

```diff
@@ -1612,30 +1838,69 @@ impl<F: Field> SubCircuit<F> for PiCircuit<F> {
                         0,
                         0,
                         0u8,
-                        &mut rpi_bytes_keccakrlc,
+                        &mut rpi_bytes_keccak_rlc,
                         challenges,
-                        &mut current_offset,
+                        &mut current_rpi_offset,
                         &mut rpi_bytes,
                         false,
                         F::ZERO,
                         zero_cell.clone(),
                     )?;
                     call_data_offset += 1;
                 }
-                assert_eq!(current_offset, 0);
+                assert_eq!(
+                    start_offset - current_rpi_offset,
+                    N_BYTES_ONE
+                        + N_BYTES_BLOCK
+                        + N_BYTES_EXTRA_VALUE
+                        + Self::Config::circuit_len_tx_id(config.max_txs)
+                        + Self::Config::circuit_len_tx_index(config.max_txs)
+                        + Self::Config::circuit_len_tx_values(config.max_txs)
+                        + config.max_calldata
+                );
+
+                // lookup assignment
+                // also assign empty to last of TxTable
+                config.assign_empty_txtable_row(&mut region, call_data_offset)?;
+
+                // assign withdrawal table and padding rows
+                let mut withdrawal_offset = call_data_offset;
```

Comment:
I think withdrawal_offset can start from 0 because it's put under wd table. tx_calldata is special case because it's appended after txs

Reply:
fixed in [86e9881](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1634/commits/86e9881d6fba72490487ec52e80af5bfa0ed3f7f)